### DISCUSSION TITLE
Custom treesitter queries

### DISCUSSION THREAD

**Discussion by enadeau** [nix-community/nixvim#1965] [Q&A] [answered]
Hi,

New user here trying to move my vim config into nixvim and I can say it's been quite fun.

I trying to use a custom folds query for python as I find the default one to be folding way too much.

The treesitter manual says:
> Nvim looks for queries as *.scm files in a queries directory under
runtimepath, where each file contains queries for a specific language and
purpose, e.g., queries/lua/highlights.scm for highlighting Lua files.
By default, the first query on runtimepath is used (which usually implies
that user config takes precedence over plugins, which take precedence over
queries bundled with Nvim).
So here I go and put in my config
```nix
extraFiles = {
    "queries/python/folds.scm".source = ./python_folds.scm;
};
```
The file indeed then show up at the expected place in `.config/nvim` but the folding behavior is not changed.
Looking at the runtime path there are a lot of nix store path showing up before my config directory.

I added to my config
```nix
extraConfigLua = ''
     vim.opt.runtimepath:prepend("/home/emilen/.config/nvim")
'';
```
and then get the my custom folding working.

The bit that surpised me is that lua/injection.scm query inserted by nix vim at the same place seems to just work fine.

I'm mostly wondering if there is a cleaner way to do that? Quite new to nix and neovim so I'm not quite sure if this is a bug and how it would be best to fix it. Shoud nixvim put the config folder first in the runtime path? I'm not sure what are the other implication of that. Maybe a option to define those query in `plugins.treesitter` would be an other idea? Maybe I'm just not seeing the right way to do that.

**Comment by siph** [comment]
Here was my solution: https://github.com/nix-community/nixvim/discussions/540#discussioncomment-6881292

**Reply by enadeau** [reply]
There are some simillarity here but python already has a parser and folding setup but I'm trying to overwrite it. The order of the runtime seems however to be preventing that

**Reply by traxys** [reply, marked as answer]
You need to add `;; extends` to the query file, see:
- https://github.com/nix-community/nixvim/blob/820f8d58eafd7121989fea3ae9e71f29699d856b/plugins/languages/treesitter/treesitter.nix#L318
- https://github.com/nix-community/nixvim/blob/main/plugins/languages/treesitter/injections.scm

**Reply by n0099** [reply]
https://github.com/nix-community/nixvim/blob/7addac6d111837217c16762968a9042eac703f7e/plugins/by-name/treesitter/injections.scm#L1
https://neovim.io/doc/user/treesitter.html#treesitter-query-modeline-extends

**Comment by i11010520** [comment]
How to use `extraFiles` exactly?

I got error when defining it in treesitter. :(
```
       error: The option `programs.nixvim.plugins.treesitter.extraFiles' does not exist. Definition values:
       - In `/nix/store/xzwca7w4hpcmpy5vjx7kl4nr56w5rxw5-source/neovim/plugins/notebook.nix':
           {
             "queries/markdown/textobjects.scm" = {
               source = /nix/store/xzwca7w4hpcmpy5vjx7kl4nr56w5rxw5-source/neovim/plugins/textobjects.scm;
             };
           }
```

**Reply by MattSturgeon** [reply]
Replace `programs.nixvim.plugins.treesitter.extraFiles` with `programs.nixvim.extraFiles`
